src/metric/definition.go

```diff
@@ -974,3 +1068,31 @@ func toCores(value definition.FetchedValue) (definition.FetchedValue, error) {
 		return nil, errors.New("error transforming to cores")
 	}
 }
+
+func fromPrometheusNumeric(value definition.FetchedValue) (definition.FetchedValue, error) {
+	switch v := value.(type) {
+	case prometheus.GaugeValue:
+		return float64(v), nil
+	case prometheus.CounterValue:
+		return float64(v), nil
+	}
+
+	return nil, fmt.Errorf("invalid type value '%v'. Expected 'gauge' or 'counter', got '%T'", value, value)
+}
+
+// Subtract returns a new FetchFunc that subtracts 2 values. It expects that the values are float64
+func Subtract(left definition.FetchFunc, right definition.FetchFunc) definition.FetchFunc {
```

Comment:
Now we understand when you said finding where to add these function was not straightforward 😅

Reply:
yeah, it's all a bit complex